### PR TITLE
Add invoice recalculation endpoint

### PR DESCRIPTION
**Pull request by XaosSintez**

### CODE REVIEW

File: spec/paths/invoices@{id}@recalculate.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  - $ref: "#/components/parameters/resourceId"
+post:
+  tags:
+    - Invoices
+  summary: Recalculate an invoice
+  operationId: PostInvoiceRecalculation
+  description: |
+    Recalculate an invoice with specified identifier string.
+  responses:
+    201:
```
Comment (ashkarpetin) [inline spec/paths/invoices@{id}@recalculate.yaml]: Should be 200?

Comment (XaosSintez) [reply spec/paths/invoices@{id}@recalculate.yaml]: No, i think it should be 201. See similar endpoints like `invoices/{id}/void`, `invoices/{id}/issue`

**Comment by 2e3s** [comment]
Missing the link in invoice resource

**Comment by XaosSintez** [comment]
@2e3s Which link?

**Comment by adamaltman** [comment]
I think he means you can add a link from the invoice itself to the recalculate operation.

**Comment by adamaltman** [comment]
(I clicked close by accident.)

File: spec/paths/invoices@{id}@recalculate.yaml (hunk 1 above)
Comment (2e3s) [reply spec/paths/invoices@{id}@recalculate.yaml]: > The HTTP 201 Created success status response code indicates that the request has succeeded and has led to the creation of a resource. The new resource is effectively created before this response is sent back and the new resource is returned in the body of the message, its location being either the URL of the request, or the content of the Location header.

File: spec/paths/invoices@{id}@recalculate.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+parameters:
+  - $ref: "#/components/parameters/resourceId"
+post:
+  tags:
+    - Invoices
+  summary: Recalculate an invoice
+  operationId: PostInvoiceRecalculation
+  description: |
+    Recalculate an invoice with specified identifier string.
```
Comment (adamaltman) [inline spec/paths/invoices@{id}@recalculate.yaml]: This description is pretty anemic.

You may want to say that this includes, discounts, shipping and taxes.

You may also want to provide some examples of when you would want to use it.  For example, if you revoke a coupon, you may want to recalculate.

**Pull request merged**